pkg/asset/installconfig/gcp/client.go

```diff
@@ -368,6 +370,24 @@ func (c *Client) getServiceUsageService(ctx context.Context) (*serviceusage.Serv
 	return svc, nil
 }
 
+// GetServiceAccount retrieves a service account from a project if it exists.
+func (c *Client) GetServiceAccount(ctx context.Context, project, serviceAccount string) (string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	svc, err := iam.NewService(ctx)
```

Comment:
I think we should not give a short timeout context when creating a client/service, and instead use it for the API call itself.
https://pkg.go.dev/cloud.google.com/go#hdr-Timeouts_and_Cancellation